File: frontend/src/api/familyHistory.ts
import { createRecordClient } from "./records";

export interface FamilyHistory {
  id: string;
  relative: string;
  condition: string;
  age_of_onset: string | null;
  notes: string | null;
}

export interface FamilyHistoryInput {
  relative: string;
  condition: string;
  age_of_onset?: string | null;
  notes?: string | null;
}

export const familyHistoryApi = createRecordClient<FamilyHistory, FamilyHistoryInput, FamilyHistoryInput>("/api/family-history");

File: backend/app/routers/family_history.py
from app.models.extended_records import FamilyHistory
from app.routers.records import build_list_router
from app.schemas.extended_records import FamilyHistoryCreate, FamilyHistoryUpdate, FamilyHistoryResponse

router = build_list_router(
    prefix="/api/family-history",
    tag="family-history",
    model=FamilyHistory,
    create_schema=FamilyHistoryCreate,
    update_schema=FamilyHistoryUpdate,
    response_schema=FamilyHistoryResponse,
)
